app/controllers/home_controller.rb

```diff
@@ -105,11 +105,32 @@ def index
     ]
 
     projects = Project.all.order(id: :asc)
-    unique_donors = projects.pluck(:donors).uniq.sort
-    #unique_category = projects.pluck(:category).uniq.sort
-    #unique_ecosystem = projects.pluck(:ecosystem).uniq.sort
-    unique_country = projects.pluck(:country).uniq.sort
-    unique_ocean_based_region = projects.pluck(:ocean_based_region).uniq.sort
+    unique_donors = Donor.pluck(:name).sort
+    unique_countries = Country.pluck(:name).sort
+    unique_ocean_based_regions = OceanRegion.pluck(:name).sort
+    unique_categories = projects.pluck(:category).uniq.sort
+    unique_ecosystems = Ecosystem.pluck(:name).sort
+
+    projects = Project.all.to_a.map! do |project|
```

Comment:
Can we create a method in the `Project` model for doing this and possibly the filters as well?
The controller should not "know" much about the logic of the data, so I think it would be best to assign this task to the model

Reply:
Sure @Levia that is a much better idea! I will create some functions in the model - do you have any suggestions for the names of these functions which will return the necessary data for the filters and the projects?

Reply:
I am not sure of the conventions of what the names should be.

Reply:
Well, there are no conventions probably for these things, but you can create something called `to_json` which will do all those things.
And you can also put those `pluck` in a function for the filters